plugins/statuslines/lualine.nix

```
@@ -1,177 +1,270 @@
-
-      package = lib.mkPackageOption pkgs "lualine" {
-        default = [
-          "vimPlugins"
-          "lualine-nvim"
-        ];
-      };
-
-      gitPackage = lib.mkPackageOption pkgs "git" {
-        nullable = true;
-      };
+              };
+            in
+            maybeRaw (listOf (either str (lib.nixvim.transitionType oldType coerceFn (maybeRaw newType))));
+          inherit description;
+        };
 
-      iconsEnabled = mkOption {
-        type = types.bool;
-        description = "Whether to enable/disable icons for all components.";
-        default = true;
+      mkEmptySectionOption = name: {
+        lualine_a = mkComponentOptions "Left section on left side.";
+        lualine_b = mkComponentOptions "Middle section on left side.";
+        lualine_c = mkComponentOptions "Right section on left side.";
+        lualine_x = mkComponentOptions "Left section on right side.";
+        lualine_y = mkComponentOptions "Middle section on right side.";
+        lualine_z = mkComponentOptions "Right section on right side.";
       };
 
-      theme = helpers.defaultNullOpts.mkNullable (
-        with types; either str attrs
-      ) "auto" "The theme to use for lualine-nvim.";
+    in
+    {
+      options = {
+        icons_enabled = defaultNullOpts.mkBool true ''
+          Whether to enable icons for all components.
 
-      componentSeparators = mkSeparatorsOption {
-        leftDefault = "";
-        rightDefault = "";
-      };
+          This option is also available on individual components to control whether they should display icons.
+        '';
 
-      sectionSeparators = mkSeparatorsOption {
-        leftDefault = "";
-        rightDefault = "";
-      };
+        theme = defaultNullOpts.mkNullable (
+          with types; either str attrs
+        ) "auto" "The theme to use for lualine-nvim.";
 
-      disabledFiletypes = {
-        statusline = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-          Only ignores the ft for statusline.
-        '';
+        component_separators = mkSeparatorsOption {
+          left = "";
+          right = "";
+        };
 
-        winbar = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-          Only ignores the ft for winbar.
-        '';
-      };
+        section_separators = mkSeparatorsOption {
+          left = "";
+          right = "";
+        };
 
-      ignoreFocus = helpers.defaultNullOpts.mkListOf types.str [ ] ''
-        If current filetype is in this list it'll always be drawn as inactive statusline and the
-        last window will be drawn as active statusline.
+        disabled_filetypes = defaultNullOpts.mkNullableWithRaw' {
+          description = ''
+            Filetypes in which to disable lualine.
+            Allows you to specify filetypes that you want to only disable on specific components.
+          '';
+          pluginDefault = { };
+          type =
+            with types;
+            either (listOf str) (submodule {
```

```suggestion
            either (listOf (maybeRaw str)) (submodule {
```

Frustrating having to be this verbose...